From: Yang Yingliang <yangyingliang@huawei.com>
Subject: mm/ksm: change break_ksm_pmd|pud_entry() to static
Date: Thu, 20 Oct 2022 15:59:13 +0800

break_ksm_pmd|pud_entry() is only used in ksm.c now, change them to
static.

Link: https://lkml.kernel.org/r/20221020075913.1046481-1-yangyingliang@huawei.com
Fixes: 16ee05ec4698 ("mm/ksm: convert break_ksm() to use walk_page_range_vma()")
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Cc: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/ksm.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/ksm.c~mm-ksm-convert-break_ksm-to-use-walk_page_range_vma-fix-2
+++ a/mm/ksm.c
@@ -420,7 +420,7 @@ static inline bool ksm_test_exit(struct
 	return atomic_read(&mm->mm_users) == 0;
 }
 
-int break_ksm_pud_entry(pud_t *pud, unsigned long addr, unsigned long next,
+static int break_ksm_pud_entry(pud_t *pud, unsigned long addr, unsigned long next,
 			struct mm_walk *walk)
 {
 	/* We only care about page tables to walk to a single base page. */
@@ -429,7 +429,7 @@ int break_ksm_pud_entry(pud_t *pud, unsi
 	return 0;
 }
 
-int break_ksm_pmd_entry(pmd_t *pmd, unsigned long addr, unsigned long next,
+static int break_ksm_pmd_entry(pmd_t *pmd, unsigned long addr, unsigned long next,
 			struct mm_walk *walk)
 {
 	bool *ksm_page = walk->private;
_
